lib/plugins.ts

```diff
@@ -436,11 +428,21 @@ export class Plugins {
    * @return {webdriver.promise.Promise|Q.Promise} A promise which resolves to the
    *     function's return value
    */
-  safeCallPluginFun(
-      pluginObj: ProtractorPlugin, funName: string, args: IArguments, promiseType: PromiseType,
-      failReturnVal: any): any {
-    var deferred = promiseType == PromiseType.Q ? Q.defer() : webdriver.promise.defer();
-    var logError = (e: any) => {
+  private safeCallPluginFun(
+      pluginObj: ProtractorPlugin, funName: string, args: any[], promiseType: PromiseType,
+      failReturnVal: any) {
+    let resultPromise: webdriver.promise.Promise<any>|q.Promise<any>;
```

Comment:
Hm, probably you're right. I searched through the [history of edits](https://github.com/kriskowal/q/wiki/API-Reference/_history) of the API reference. It turns out it never was documented. So we can't really call it deprecated.